ProcessMaker/processmaker
# Improve Guided Template URL Generation

This PR improves the generated URL for guided templates by ensuring the `template` URL parameter is consistent and utilizes the `unique_template_id` property established in the `wizard-template-details.xml` file located in the Wizard Templates GitHub repo.

## Solution

- Creating a migration to add a `new unique_template_id` column to the `wizard_templates` table.
- Updating the `sync-wizard-templates` job to map the `unique_template_id` value from the `wizard-template-details.xml` file.
- Updating the ID being referenced in the `template` URL parameter.

## How to Test

1. Run `php artisan migrate`.
2. Check the database.
3. Ensure the new `unique_template_id` column exists on the `wizard_templates` table.
4. Run `php artisan processmaker:sync-wizard-templates`.
5. Check the database.
6. Ensure the template was properly synced with the `unique_template_id` value being set according to the `wizard-template-details.xml` file.
7. In the UI, navigate to Processes > Guided Templates.
8. Ensure the template is visible.
9. Select the template.
10. Ensure the URL `template` parameter matches that of the `unique_template_id`.
11. Copy the URL
12. Open a new tab
13. Paste the URL
14. Ensure the template modal appears

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-13277

ci:next

.

## Code Review Checklist
